### ISSUE TITLE
OutdatedGumpyException

### ISSUE THREAD

**Issue by alantsangmb** [oxfordmmm/gnomonicus#28]
Hello, I have updated gnomonicus to the latest version, v2.1.0, and I have regenerated the genome object again using gbkToPkl.
However, when I run gnomonicus, I receive an OutdatedGumpyException error, and the log file shows the following messages:

02-Aug-23 01:09:02 -  WARNING - No outputs selected. No files will be created
02-Aug-23 01:09:09 -  ERROR - Genome object is outdated!
02-Aug-23 01:09:09 -  ERROR - Genome object is outdated!

It seems that I am using an outdated genome object. Is it possible that I missed a step or made a mistake during the update process?
Thank you.

**Comment by alantsangmb** [comment]
I ran gnomonicus again, but this time I passed the GenBank file NC_000962.3.gbk instead of the NC_000962.3.gbk.pkl file and let gnomonicus generate the pkl file itself. It appears that Gnomonicus completed successfully, so I plan to use this pkl file for future runs. Thank you so much. Here is the partial message from the log file:

02-Aug-23 01:45:46 -  INFO - gnomonicus starting with output directory /MTB/clockwork_results/debug
02-Aug-23 01:45:46 -  WARNING - No outputs selected. No files will be created
02-Aug-23 01:45:46 -  DEBUG - Using file Ref/NC_000962.3.gbk
02-Aug-23 01:45:46 -  INFO - Path was not to a gzipped file. Defaulting to normal reading
02-Aug-23 01:45:46 -  INFO - Genome object not a pickle, checking if pickled version exists. Error: invalid literal for int() with base 10: 'OCUS       NC_000962            4411532 bp    DNA     linear   CON 14-DEC-2017\n'
02-Aug-23 01:45:49 -  INFO - Path was to a gzipped file. Decompressing...
02-Aug-23 01:45:53 -  INFO - Genome object is outdated! Trying with the original filepath
02-Aug-23 01:46:57 -  DEBUG - Loaded reference genome
02-Aug-23 01:47:03 -  DEBUG - Applied the VCF to the reference
02-Aug-23 01:47:04 -  DEBUG - Loaded resistance catalogue
02-Aug-23 01:49:41 -  DEBUG - Got the genome difference
02-Aug-23 01:49:42 -  DEBUG - Populated and saved variants.csv

##########

02-Aug-23 01:51:35 -  DEBUG - Populated and saved mutatons.csv
02-Aug-23 01:51:35 -  DEBUG - Populated and saved effects.csv
02-Aug-23 01:51:35 -  INFO - ********** Successfully completed **********
02-Aug-23 01:51:35 -  INFO - VCF file: Var_call_TH23003474_out/final.vcf
02-Aug-23 01:51:35 -  INFO - Reference genome file: Ref/NC_000962.3.gbk
02-Aug-23 01:51:35 -  INFO - Catalogue reference genome: NC_000962.3
02-Aug-23 01:51:35 -  INFO - Catalogue name: WHO-UCN-GTB-PCI-2021.7
02-Aug-23 01:51:35 -  INFO - Catalogue version: 1.0
02-Aug-23 01:51:35 -  INFO - Catalogue grammar: GARC1
02-Aug-23 01:51:35 -  INFO - Catalogue values: ['R', 'U', 'S']
02-Aug-23 01:51:35 -  INFO - Catalogue path: Ref/NC_000962.3_WHO-UCN-GTB-PCI-2021.7_v1.0_GARC1_RUS.csv
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_AMI S
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_CAP S
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_DLM S
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_EMB U
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_ETH S
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_INH R
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_KAN S
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_LEV S
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_LZD S
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_MXF S
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_PZA S
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_RIF R
02-Aug-23 01:51:35 -  INFO - WGS_PREDICTION_STM S
02-Aug-23 01:51:35 -  INFO - Completed in 348.72469997406006s

**Comment by JeremyWesthead** [comment]
This is due to changes within our genetic library `gumpy` causing issues. Essentially we encountered an issue where a previously pickled `gumpy.Genome` object was instantiated with an older `gumpy` version; once unpickled, it retains the functionality of the older `gumpy` version - causing issues when relying on newer `gumpy` functions.

To combat this, I added a check to ensure that the version of `gumpy` used to instantiate the pickle matches the version of `gumpy` currently being used. 

*Going forwards, this will be important to note during any `gnomonicus` update which changes the `gumpy` version.*
I'll update the README to ensure this is noted
